x/feedist/handler.go

```diff
@@ -189,8 +255,8 @@ func distribute(db weave.KVStore, ctrl CashController, source weave.Address, rec
 
 		for _, r := range recipients {
 			amount := x.Coin{
-				Whole:      (c.Whole / chunks) * int64(r.Weight),
-				Fractional: (c.Fractional / chunks) * int64(r.Weight),
+				Whole:      (c.Whole / chunks) * int64(r.Weight/div),
```

Comment:
How about this?

```go
// Split divides the value of a coin into given amount of pieces and returns a
// single piece.
// Whole to fractional relation differs between currencies and therefore it
// must be passed explicitely.
// It might be that precise splitting is not possible. If so, any leftover of a
// fractional value is ignored. For example splitting 4 EUR into 3 pieces will
// result in a single piece being 1.33 EUR and 1 cent being ignored
// 4 = 1.33 x 3 + 1
func (c Coin) Split(pieces int64, wholeToFractional int64) Coin {
	fractional := c.Fractional / pieces

	// When dividing whole and there is a leftover then convert it to
	// fractional and split as well.
	if leftover := c.Whole % pieces; leftover != 0 {
		fractional += (leftover * wholeToFractional) / pieces
	}

	return Coin{
		Ticker:     c.Ticker,
		Whole:      c.Whole / pieces,
		Fractional: fractional,
	}
}
```
To use it in the context of this functionality, it would be:

```go
wholeToFrac := guessWholeToFrac(c.Ticker)
amount := c.Split(r.Weight/div, wholeToFrac) * chunks
```

Reply:
Looks fine.. but there is no `*` operator for `Coin`

This is basically `Divide`, but yeah, good.
Fine to make a Multiply function as well.

Reply:
As to `wholeToFractional`, this is fixed for all coins in weave: `x.FracUnit`
https://github.com/iov-one/weave/blob/master/x/coin.go#L12-L24

These constants are used for all operations, such as addition now, so you can just rely on them being, well, constant.

https://github.com/iov-one/weave/blob/master/x/coin.go#L54-L69
https://github.com/iov-one/weave/blob/master/x/coin.go#L192-L222

Reply:
Implemented in https://github.com/iov-one/weave/pull/316/commits/a9f849ab0dd27c96e823b4f36ce1c7a5eb47e147

I was not sure what to do when `pieces` value is invalid, so I am returning [zero value](https://github.com/iov-one/weave/pull/316/commits/a9f849ab0dd27c96e823b4f36ce1c7a5eb47e147#diff-2a10d0bf5e4f9e87e59d8885f76a0278R365). :man_shrugging: 

Use will be
```go
one, _ := c.Split(r.Weight/div)
amount := one.Multiply(chunks)
```
Now I have realized I must implement `Multiply` as well :sweat_smile:

Reply:
Now the `distribute` function has [a proper division granularity implemented](https://github.com/iov-one/weave/pull/316/commits/052495980bb0f19180e18fbaaccdbe5df10c9793)! :tada: